### PR TITLE
[docs] Communication subsystem docs update for MDL-78551

### PR DESCRIPTION
**Pull request by safatshahin**
Some more API's added as a part of MDL-78551, adding docs for them.

### CODE REVIEW

File: docs/apis/subsystems/communication/index.md [hunk 1]
```diff
@@ -149,6 +157,35 @@ This is destructive and might remove all the messages and other data associated
 
 :::
 
+### Create and configure a room according to the provider
+
+There are cases where the provider is changed for a communication instance. For example, previously the provider was set to _Provider A_ and now it has changed to _Provider B_. In this case, the room and its members need to be configured according to the new provider. Without any extra logic needed, the method `$communication->configure_room_and_membership_by_provider()` takes care of this in the following way:
+
+1. Communication provider is changed from _Provider A_ to _Provider B_.
+2. New room is created in _Provider A_.
```
Comment (sarjona) [inline docs/apis/subsystems/communication/index.md:165]: Is this correct? From the example I understood that Provider B was the new one, so I find it weird creating a new room in Provider A...

File: docs/apis/subsystems/communication/index.md [hunk 2]
```diff
@@ -32,6 +32,14 @@ Please follow the steps to enable the feature.
 
 :::
 
+## Hooks
+
+The Communication API takes advantage of the [Hooks API](../../plugintypes/communication/index.md) with actions performed in the following way:
+
+1. Actions in core have their own hook dispatched (e.g. enrol, change role, add to group).
```
Comment (sarjona) [inline docs/apis/subsystems/communication/index.md:39]:
```suggestion
1. Actions in core have their own hook dispatched (for example, enrol, change role, add to group).
```

File: docs/apis/subsystems/communication/index.md [hunk 3]
```diff
@@ -32,6 +32,14 @@ Please follow the steps to enable the feature.
...
+2. Hooks are then registered to a particular callback (`lib/db/hooks.php`).
```
Comment (sarjona) [inline docs/apis/subsystems/communication/index.md:40]: As this is a generic location where more hooks will be stored (not just the ones affecting communication), shouldn't it be mentioned explaining how to identify the communication ones? ;-)

**Review by sarjona — CHANGES_REQUESTED** [review]
Hi @safatshahin!
Thanks for improving the communication API docs to include changes implemented in https://tracker.moodle.org/browse/MDL-78551
As you'll see, I left a few points that should be clarified before merging this patch. TIA!

File: docs/apis/subsystems/communication/index.md (hunk 1 above)
Comment (safatshahin) [reply docs/apis/subsystems/communication/index.md:170]: A and B typo, resolved.

File: docs/apis/subsystems/communication/index.md (hunk 3 above)
Comment (safatshahin) [reply docs/apis/subsystems/communication/index.md:40]: Hi @sarjona 
I didn't create any communication-specific hooks, for example, hooks related to enrolment were, before enrolment, after enrolment, they were all generic. 
I have added a forth point to mention that.

**Comment by safatshahin** [comment]
Hi @sarjona 
Thank you for your comments. I have resolved all of them and made the changes.
Please have a look now.
Cheers!

**Review by sarjona — APPROVED** [review]
HI @safatshahin!
Thanks for reviewing and considering my comments :-) 
I'm happy to approve and merge this :-)

**Pull request merged**